pallets/foreign-investments/src/lib.rs

```diff
@@ -422,35 +302,473 @@ pub mod pallet {
+						T::TokenSwaps::update_order(
+							who.clone(),
+							swap_id,
+							swap.amount_in.ensure_add(new_swap.amount_in)?,
+							T::BalanceRatio::one(),
+						)?;
+
+						Ok(SwapStatus {
+							swapped: T::Balance::zero(),
+							pending: new_swap.amount_in,
+						})
+					} else {
+						let inverse_swap = swap;
+						let new_swap_amount_out = Self::amount_used_to_swap(&new_swap)?;
+
+						match inverse_swap.amount_in.cmp(&new_swap_amount_out) {
+							Ordering::Less => {
+								T::TokenSwaps::update_order(
+									who.clone(),
+									swap_id,
+									inverse_swap.amount_in.ensure_sub(new_swap_amount_out)?,
+									T::BalanceRatio::one(),
+								)?;
+
+								Ok(SwapStatus {
+									swapped: new_swap.amount_in,
+									pending: T::Balance::zero(),
+								})
+							}
+							Ordering::Equal => {
+								T::TokenSwaps::cancel_order(swap_id.clone())?;
+								Self::unregister_swap(&who, investment_id, kind, swap_id);
+
+								Ok(SwapStatus {
+									swapped: new_swap.amount_in,
+									pending: T::Balance::zero(),
+								})
+							}
+							Ordering::Greater => {
+								T::TokenSwaps::cancel_order(swap_id.clone())?;
+								Self::unregister_swap(&who, investment_id, kind, swap_id);
+
+								let inverse_swap_amount_out =
+									Self::amount_used_to_swap(&inverse_swap)?;
+
+								let amount_to_swap =
+									new_swap.amount_in.ensure_sub(inverse_swap_amount_out)?;
+
+								let swap_id = T::TokenSwaps::place_order(
+									who.clone(),
+									new_swap.currency_in,
+									new_swap.currency_out,
+									amount_to_swap,
+									T::BalanceRatio::one(),
+								)?;
+								Self::register_swap(&who, investment_id, kind, swap_id);
+
+								Ok(SwapStatus {
+									swapped: inverse_swap_amount_out,
+									pending: amount_to_swap,
+								})
+							}
+						}
+					}
+				}
+			}
+		}
+	}
+
+	impl<T: Config> ForeignInvestment<T::AccountId> for Pallet<T> {
+		type Amount = T::Balance;
+		type CurrencyId = T::CurrencyId;
+		type Error = DispatchError;
+		type InvestmentId = T::InvestmentId;
+
+		fn increase_foreign_investment(
+			who: &T::AccountId,
+			investment_id: T::InvestmentId,
+			foreign_amount: T::Balance,
+			foreign_currency: T::CurrencyId,
+			pool_currency: T::CurrencyId,
+		) -> DispatchResult {
+			ForeignInvestmentInfo::<T>::insert(
+				who,
+				investment_id,
+				Info {
+					foreign_currency,
+					pool_currency,
+				},
+			);
+
+			let pool_amount = T::CurrencyConverter::stable_to_stable(
+				pool_currency,
+				foreign_currency,
+				foreign_amount,
+			)?;
+
+			let status = Self::apply_swap(
+				who,
+				investment_id,
+				ForeignSwapKind::Investment,
+				Swap {
+					currency_in: pool_currency,
+					currency_out: foreign_currency,
+					amount_in: pool_amount,
+				},
+			)?;
+
+			T::Investment::update_investment(
+				who,
+				investment_id,
+				T::Investment::investment(who, investment_id)?.ensure_add(status.swapped)?,
```

Comment:
no change of order will fail on the investments side...

Reply:
At least in the extrinsics. not sure about the trait

Reply:
Thanks for the reminder!